### DISCUSSION TITLE
Function for retrieving parent xpath

### DISCUSSION THREAD

**Discussion by nbiskup-eridan** [sysrepo/sysrepo#3452] [Q&A]
Hi, 
is there a function available that enables retrieving the parent xpath, similar to the sr_xpath_node_name function in sysrepo/xpath.h?

Here’s an example of what I’m looking for:

Input:
```
/o-ran-uplane-conf:user-plane-configuration/static-low-level-tx-endpoints
```
Expected Output:
```
/o-ran-uplane-conf:user-plane-configuration
```

**Comment by michalvasko** [comment]
No, XPath processing functions are not public in *libyang*. But it should be fairly easy to write such a function yourself.

**Comment by nbiskup-eridan** [comment]
Thank you for your response. Writing it independently is not a problem, but I believe it should be part of the section that handles such functionalities. Therefore, I think it could be added to sysrepo/xpath.h?

**Reply by michalvasko** [reply]
You are free to provide a patch. The functions in `sysrepo/xpath.h` are present mainly for backwards compatibility because they are relatively slow so we are not using them nor improving them.
